### DISCUSSION TITLE
Table panel should allow to select fields to display & re-order columns

### DISCUSSION THREAD

**Discussion by cykl** [grafana/grafana#43479] [Dashboards, Panels & Visualizations]
Let's say that I use an Elasticsearch data source. This data source allows me to filter documents but not to select which document fields will be returned (I can only select _raw data_ or _logs_ metric type). 

Now, I want to display the result using a table panel. With the old table panel, I was able to select which fields to display and in which order (annoyingly because you had to erase and redo everything to change the order).

New table panel doesn't have such feature.

You can use override _Fields with name matching regex_ and enable _hide in table_ to hide all unwanted fields. But I believe most people want opt-in rather than opt-out (e.g. when a new field is added please do not destroy my carefully crafted dashboard).

You cannot re-order columns at all.

It would also be nice to let dashboard viewer to easily hide field / show new fields (ex. panel is configured to display most useful fields but a viewer could be interested to display another one).

**Comment by AbelThorne** [comment]
I believe you are supposed to use Transformation to do that. 
There is a Transformation called "Organize fields" that will allow you to select the fields you want to display or not, how to rename them and also reorder them. The upside is that this will work regardless of the panel you choose to display the data, table or anything else. 
Hopefully this will fix your need

**Reply by JykkeDaMan** [reply]
Is this Transformation possible for the end user viewing the dashboard or admin creating the dashboard only?
I would need as an dashboard viewer to be able to select the table fields and order dynamically when needed.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
